### PR TITLE
Ok 48 remove movie from list

### PR DESCRIPTION
**Pull request by dufourmax1738**
# Jira Link:

[Jira Link ](https://champlainsaintlambert.atlassian.net/browse/OK-48?atlOrigin=eyJpIjoiMzJiZjMyNzM4Y2I3NDI5OWI0MWU2MTM4YmNlYmVhYzEiLCJwIjoiaiJ9)

# Context

Users can now remove movies from lists from their account page

# Changes

- Changed movie tile so that if it is in the profile page it has an X button which deletes that movie from the users list
- Refactored code from movieTile, profileMovieList, and users page so that it calls it from the firebase file
- Added User and ListedMovie types
- Changed Movie tile styling so info appears on hover of the whole card (not just the link)
- Changed profile page so that movies update when a movie is removed
- Added an api call which adds dummy list data to the test user

# Before and After UI (Required for UI-Impacting PRs) 

[Before: ]()
image

[After: ]()
image

### CODE REVIEW

**Review by ClementCadieux — APPROVED** [review]
Looks all good, nice job. I like the overlay effect.

**Comment by ClementCadieux** [comment]
Can you tell me why the package-lock.json file seems to have lost 15 thousand lines?

**Comment by dufourmax1738** [comment]
> Can you tell me why the package-lock.json file seems to have lost 15 thousand lines?

I am not sure why so many lines got removed, it happened when I tried updating firebase in our project because the functions we had available did not match the current documentation but it had no effect in the end. Everything still works but now we are using the most recent version.

File: components/ProfileMovieList.tsx [hunk 1]
```diff
@@ -1,45 +1,29 @@
...
     const [finalMovies, setMovies] = useState([])
 
+    useEffect(()=>{
+            fetchMovieDetailsForList(props.movies)
+                .then((res) => {
+                    // @ts-ignore
+                    setMovies(res)
+                    console.log(finalMovies)
+                })
+    },[props.movies])
```
Comment (DuncanBH) [inline components/ProfileMovieList.tsx:25]: Clever restructuring, this seems much more efficient.

File: lib/firebase.ts [hunk 2]
```diff
@@ -29,7 +29,62 @@ export function docToJSON(doc: DocumentSnapshot) {
...
+
+
+
+    // console.table(userInfo);
+    // @ts-ignore
+    return (await getDocs(searchQuery)).docs.map(docToJSON);
+}
+
 export async function getMovieById(id : number) {
+    console.log("DB READ!")
```
Comment (DuncanBH) [inline lib/firebase.ts:87]: Be careful with the console.logs. These can potentially expose information to the user, and will cause slowdowns in your code.

File: pages/api/DummyListedMovies.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                alert("Entire Document has been updated successfully");
+            })
+            .catch(error => {
+                alert(error);
+            })
+        return res.status(200).json({ test: 'posted' })
+    } else {
+        res.status(200).json({ test: 'works' })
+    }
+}
```
Comment (DuncanBH) [inline pages/api/DummyListedMovies.tsx:22]: Why is this an endpoint? 
From what I am understanding, anyone who makes  a POST request to the /api/DummyListedMovies endpoint will cause it to add these movies to a hardcoded account?

**Review by DuncanBH — APPROVED** [review]
LGTM except the API endpoint. I do not understand what it is for or why it is there.

Good work on the CSS effect, that was well constructed 👍

File: lib/firebase.ts (hunk 2 above)
Comment (dufourmax1738) [reply lib/firebase.ts:87]: I am aware of that but we have had many major issues where the DB is called way too often so I will be leaving that in for now so we do not get kicked out of the DB again

File: pages/api/DummyListedMovies.tsx (hunk 3 above)
Comment (dufourmax1738) [reply pages/api/DummyListedMovies.tsx:22]: that is correct, this is not meant to be a permanent fixture of the application but as a way to quickly add data for testing. I have a text file with all the things we will need to remove before releasing the final version.

**Pull request merged**